help.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"os"
+	"strings"
+)
+
+// runHelp implements the 'help' command.
+func runHelp(args []string) int {
+	if len(args) == 0 {
+		mainUsage(os.Stdout)
+		return 0
+	}
+	if len(args) != 1 {
+		fmt.Fprintf(os.Stderr, "modcop help %s: too many arguments\n", strings.Join(args, " "))
+		return 2
+	}
+	t := template.Must(template.New("").Parse(commandHelpTemplate))
+	for _, c := range commands {
+		if c.Name() == args[0] {
+			if err := t.Execute(os.Stdout, c); err != nil {
+				errorf("cannot write usage output: %v", err)
+			}
+			return 0
+		}
+	}
+	fmt.Fprintf(os.Stderr, "gohack help %s: unknown command\n", args[0])
+	return 2
+}
+
+func mainUsage(f io.Writer) {
+	t := template.Must(template.New("").Parse(mainHelpTemplate))
+	if err := t.Execute(f, commands); err != nil {
+		errorf("cannot write usage output: %v", err)
+	}
+}
+
+var mainHelpTemplate = `
+The modcop command checks that module dependencies do not stray outside
+well defined boundaries. It expects that the set of allowed dependencies
+will be checked in along with the source code, and modcop will be called
+as part of a CI check, so any addition to the set of allowed dependencies
+will need to be explicitly vetted as part of the code review process.
+
+The set of dependencies checked by modcop does not include dependencies
+of test code outside of the module being checked - it differs from
```

Comment:
I think you need to define therefore that the packages argument is limited to those within the main module? `go generate` does a similar thing.

Reply:
This can then be refined to say "outside of the main module"

Reply:
I actually realised that it was more accurate to say "outside of the packages being checked", because the granularity is package-level not module-level.